chunky/src/java/se/llbit/chunky/entity/BeaconBeam.java

```diff
@@ -0,0 +1,136 @@
+package se.llbit.chunky.entity;
+
+import se.llbit.chunky.resources.Texture;
+import se.llbit.chunky.world.Material;
+import se.llbit.chunky.world.material.TextureMaterial;
+import se.llbit.json.Json;
+import se.llbit.json.JsonObject;
+import se.llbit.json.JsonValue;
+import se.llbit.math.*;
+import se.llbit.math.primitive.Primitive;
+import se.llbit.nbt.CompoundTag;
+import se.llbit.util.JsonUtil;
+
+import java.util.Collection;
+import java.util.LinkedList;
+
+public class BeaconBeam extends Entity implements Poseable {
+
+  public static final Material beaconBeamMaterial = new TextureMaterial(Texture.beaconBeam);
+
+  private static final Quad[] beam = new Quad[]{
+      new Quad(
+          new Vector3(5 / 16.0, 0.0, 5 / 16.0),
+          new Vector3(5 / 16.0, 0.0, 11 / 16.0),
+          new Vector3(5 / 16.0, 1.0, 5 / 16.0),
+          new Vector4(1.0, 0.0, 1.0, 0.0)
+      ),
+      new Quad(
+          new Vector3(11 / 16.0, 0.0, 11 / 16.0),
+          new Vector3(11 / 16.0, 0.0, 5 / 16.0),
+          new Vector3(11 / 16.0, 1.0, 11 / 16.0),
+          new Vector4(1.0, 0.0, 1.0, 0.0)
+      ),
+      new Quad(
+          new Vector3(5 / 16.0, 1.0, 5 / 16.0),
+          new Vector3(11 / 16.0, 1.0, 5 / 16.0),
+          new Vector3(5 / 16.0, 0.0, 5 / 16.0),
+          new Vector4(1.0, 0.0, 1.0, 0.0)
+      ),
+      new Quad(
+          new Vector3(11 / 16.0, 1.0, 11 / 16.0),
+          new Vector3(5 / 16.0, 1.0, 11 / 16.0),
+          new Vector3(11 / 16.0, 0.0, 11 / 16.0),
+          new Vector4(1.0, 0.0, 1.0, 0.0)
+      )
+  };
+
+  private final JsonObject pose;
+  private double scale = 1;
+  private int height = 256;
+
+  public BeaconBeam(Vector3 position) {
+    super(position);
+    this.pose = new JsonObject();
+    pose.add("all", JsonUtil.vec3ToJson(new Vector3(0, 0, 0)));
```

Comment:
What I like about the `height` attribute is that you can make the beams _not_ stop at a certain height. But maybe you could calculate the default height here with 255 - position.y (please check that formula for off-by-one errors :smile:) so that all beacons are as high as ingame by default?

Reply:
I can change that. Beacons in game do go up infinitely, though.

Reply:
No need to change this then